libbeat/reader/syslog/rfc3164_test.go

```diff
@@ -24,123 +24,212 @@ import (
-		WantErr: ErrTimestamp,
-	},
-	"err-hostname-too-long": {
-		In:      "<13>Oct 11 22:14:15 abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz12345.abcdefghijklmnopqrstuvwxyz1234567 this is the message",
-		WantErr: ErrHostname,
-	},
-}
-
 func TestParseRFC3164(t *testing.T) {
-	for name, tc := range parseRFC3164Cases {
+	tests := map[string]struct {
+		In      string
+		Want    message
+		WantErr string
+	}{
+		"ok": {
+			In: "<13>Oct 11 22:14:15 test-host this is the message",
+			Want: message{
+				timestamp: mustParseTime(time.Stamp, "Oct 11 22:14:15", time.Local),
+				priority:  13,
+				facility:  1,
+				severity:  5,
+				hostname:  "test-host",
+				msg:       "this is the message",
+			},
+		},
+		"ok-rfc3339": {
+			In: "<13>2003-08-24T05:14:15.000003-07:00 test-host this is the message",
+			Want: message{
+				timestamp: mustParseTime(time.RFC3339Nano, "2003-08-24T05:14:15.000003-07:00", nil),
+				priority:  13,
+				facility:  1,
+				severity:  5,
+				hostname:  "test-host",
+				msg:       "this is the message",
+			},
+		},
+		"ok-process": {
+			In: "<13>Oct 11 22:14:15 test-host su: this is the message",
+			Want: message{
+				timestamp: mustParseTime(time.Stamp, "Oct 11 22:14:15", time.Local),
+				priority:  13,
+				facility:  1,
+				severity:  5,
+				hostname:  "test-host",
+				process:   "su",
+				msg:       "this is the message",
+			},
+		},
+		"ok-process-pid": {
+			In: "<13>Oct 11 22:14:15 test-host su[1024]: this is the message",
+			Want: message{
+				timestamp: mustParseTime(time.Stamp, "Oct 11 22:14:15", time.Local),
+				priority:  13,
+				facility:  1,
+				severity:  5,
+				hostname:  "test-host",
+				process:   "su",
+				pid:       "1024",
+				msg:       "this is the message",
+			},
+		},
+		"non-standard-date": {
+			In: "<123>Sep 01 02:03:04 hostname message",
+			Want: message{
+				timestamp: mustParseTime(time.Stamp, "Sep 1 02:03:04", time.Local),
+				priority:  123,
+				facility:  15,
+				severity:  3,
+				hostname:  "hostname",
+				msg:       "message",
+			},
+		},
+		"err-pri-not-a-number": {
+			In: "<abc>Oct 11 22:14:15 test-host this is the message",
+			Want: message{
+				timestamp: mustParseTime(time.Stamp, "Oct 11 22:14:15", time.Local),
+				priority:  -1,
+				hostname:  "test-host",
+				msg:       "this is the message",
+			},
+			WantErr: `validation error at position 2: invalid priority: strconv.Atoi: parsing "abc": invalid syntax`,
+		},
+		"err-pri-out-of-range": {
+			In: "<192>Oct 11 22:14:15 test-host this is the message",
+			Want: message{
+				timestamp: mustParseTime(time.Stamp, "Oct 11 22:14:15", time.Local),
+				priority:  -1,
+				hostname:  "test-host",
+				msg:       "this is the message",
+			},
+			WantErr: ErrPriority.Error(),
```

Comment:
Is it worth making use of the validation error type here? The `err` field would by given an `ErrPriority` to hold.